### PR TITLE
feat: storage details route

### PR DESCRIPTION
**Pull request by Defferrard**

### CODE REVIEW

File: src/services/item/repository.ts [hunk 1]
```diff
@@ -644,6 +650,48 @@ export class ItemRepository {
...
+      updatedAt: item.item_updated_at,
+      size: JSON.parse(item.item_extra)[itemType].size,
+      path: JSON.parse(item.item_extra)[itemType].path,
+      parent: item.parent_id
+        ? {
+            id: item.parent_id,
+            name: item.parent_name,
+          }
+        : undefined,
+    }));
```
Comment (ReidyT) [inline src/services/item/repository.ts:690]: Should we define a type for that ?

File: src/services/member/controller.test.ts [hunk 2]
```diff
@@ -185,4 +214,225 @@ describe('Member Controller', () => {
...
+    });
+    it('returns bad request when page size is lower than 1', async () => {
+      const response = await app.inject({
+        method: HttpMethod.Get,
+        url: '/members/current/storage/files',
+        query: { pageSize: FILE_METADATA_MIN_PAGE_SIZE - 1 + '' },
+      });
+      expect(response.statusCode).toBe(StatusCodes.BAD_REQUEST);
+    });
+    it('returns bad request when page size is greater than 50', async () => {
```
Comment (ReidyT) [inline src/services/member/controller.test.ts:241]: Not sure if it is working:

```suggestion
    it(`returns bad request when page size is greater than ${FILE_METADATA_MAX_PAGE_SIZE}`, async () => {
```

File: src/services/member/controller.test.ts [hunk 3]
```diff
@@ -185,4 +214,225 @@ describe('Member Controller', () => {
...
+      const response = await app.inject({
+        method: HttpMethod.Get,
+        url: '/members/current/storage/files',
+        query: { pageSize: FILE_METADATA_MAX_PAGE_SIZE + 1 + '' },
+      });
+      expect(response.statusCode).toBe(StatusCodes.BAD_REQUEST);
+    });
+
+    it('without parent', async () => {
```
Comment (ReidyT) [inline src/services/member/controller.test.ts:250]: Maybe we can be more explicit in the test's name ?

File: src/services/member/controller.test.ts [hunk 4]
```diff
@@ -185,4 +214,225 @@ describe('Member Controller', () => {
...
+        for (let i = 0; i < totalFiles; i++) {
+          await saveFile(member, rootFile);
+        }
+
+        for (let i = 0; i < 5; i++) {
+          await saveFile(anotherMember);
+        }
+      });
+
+      it('default', async () => {
```
Comment (ReidyT) [inline src/services/member/controller.test.ts:288]:
```suggestion
      it('default parameters when not specified', async () => {
```

File: src/services/member/controller.ts [hunk 5]
```diff
@@ -51,6 +59,32 @@ const controller: FastifyPluginAsync = async (fastify) => {
...
+        return;
+      }
+      const member = notUndefined(user?.member);
+      return {
+        ...(await storageService.getStorageFilesMetadata(
+          member,
+          fileService.fileType,
+          { page, pageSize },
+          buildRepositories(),
+        )),
```
Comment (ReidyT) [inline src/services/member/controller.ts:82]: Maybe this is more readable ?

```suggestion
      const storageFilesMetadata = await storageService.getStorageFilesMetadata(
          member,
          fileService.fileType,
          { page, pageSize },
          buildRepositories(),
        );
      return {
        ...storageFilesMetadata,
```

**Review by ReidyT — APPROVED** [review]
Thanks for this PR, LGTM ! I just let some comments 😄

File: src/services/member/controller.test.ts (hunk 2 above)
Comment (Defferrard) [reply src/services/member/controller.test.ts:241]: Having variables in the test name makes it difficult to be runned independently from command line or Jest Runner.
What about `returns bad request when page size is greater than the maximum` ?

Comment (ReidyT) [reply src/services/member/controller.test.ts]: Okay I understand. Yeah I think it's good :). It was okay with the `50`, my only issue is if we change the limit we have to remember to change the test name too. Not a big deal, so it's as you want.

Comment (Defferrard) [reply src/services/member/controller.test.ts]: Yeah, I agree with you about the number!

File: package.json [hunk 6]
```diff
@@ -60,7 +60,7 @@
     "@fastify/static": "7.0.4",
     "@fastify/websocket": "7.2.0",
     "@graasp/etherpad-api": "2.1.1",
-    "@graasp/sdk": "4.22.0",
+    "@graasp/sdk": "graasp/graasp-sdk#611-add-file-metadata-type-for-storage-route",
```
Comment (pyphilia) [inline package.json:63]: Update!

File: src/services/member/controller.ts [hunk 7]
```diff
@@ -51,6 +59,33 @@ const controller: FastifyPluginAsync = async (fastify) => {
...
+        reply.status(StatusCodes.BAD_REQUEST).send();
+        return;
+      }
+      const member = notUndefined(user?.member);
+      const storageFilesMetadata = await storageService.getStorageFilesMetadata(
+        member,
+        fileService.fileType,
+        { page, pageSize },
+        buildRepositories(),
+      );
```
Comment (pyphilia) [inline src/services/member/controller.ts:81]: I usually try to keep `actor` as first arg, then `repositories` as 2nd arg for some kind of consistency.

File: src/services/member/schemas.ts [hunk 8]
```diff
@@ -121,6 +121,54 @@ export const getStorage: FastifySchema = {
   },
 };
 
+export const getStorageFiles: FastifySchema = {
+  querystring: {
+    type: 'object',
+    properties: {
+      page: { type: 'number', default: 1 },
+      pageSize: { type: 'number', default: 10 },
```
Comment (pyphilia) [inline src/services/member/schemas.ts:129]: The default here is not the same as in `src/services/member/constants.ts`? Should we use the constants here?

**Review by pyphilia — APPROVED** [review]
LGTM, I've added some minor comments, thanks for this PR! 📗

**Review by mariembencheikh — APPROVED** [review]
Good work ! Thank you !

**Pull request merged**